### DISCUSSION TITLE
Determination of marteniste variants

### DISCUSSION THREAD

**Discussion by Achintya-Patra** [mtex-toolbox/mtex#1843] [Ask Anything]
I have been trying to implement the martensite variant calculation example given in the documentation. Although the code runs fine when I am calculating and plotting the variant IDs and packet IDs but the bain ID option is not working. 
```matlab
[variantId, packetId, bainId] = calcVariantId(parenOri,childOri,p2cIter,'morito');
```
So when I am using the aforementioned line of code, I am getting the following error
```
Error using calcVariantId
Too many output arguments.
```
However, the error does not occur when I am calculating the packet IDs. Addition of the extra output argument is causing the error. Has this output argument been removed from calcVariantId command or I am doing some mistake? The rest of the code is same as given in the example file and is working fine. I just wanted to plot the bain variants in my microstructure.

**Comment by ralfHielscher** [comment]
You need to run the newest MTEX version 5.10. to use this feature.

Ralf.

**Reply by Achintya-Patra** [reply]
Thank you Dr Hielscher. I am currently using the version 5.9, is the version 5.10 going to available soon? 
In order to get the distribution of Bain variants, I am taking the variant Id list and then grouping together the variant Ids which belong to a common Bain group. Is this is proper way to do it? can this method be somehow further extended to plot in a map with a different color assigned for each bain group (like the availble option to plot packet Ids)

**Reply by nyyssont** [reply]
Hello,

there is a typo on the main MTEX website. The top option on the news looks like a duplicate of MTEX 5.9.0 but actually is version MTEX 5.10.0

**Reply by Achintya-Patra** [reply]
Oh, I didn't know that. I thought 5.9 is the latest version. Thanks a lot for this information Dr. Nyyssönen.
